### PR TITLE
BED-5641

### PR DESCRIPTION
**Pull request by kpowderly**
## Description

Added contract tests for several auth API handlers including SAML, OIDC, and SSO login endpoints. Created comprehensive tests for handlers by following established contract testing patterns.

Note: This ticket is marked as a draft while I resolve final tests.

1. GetPermission
2. GetRole
3. GetUser
4. GetSelf
5. CreateAuthToken
6. DeleteAuthToken
7. EnrollMFA
8. Logout
9. OIDCLoginHandler
10. OIDCCallbackHandler
11. SAMLLoginRedirect
12. SAMLCallbackRedirect
13. ListSAMLSignOnEndpoints
14. ListSAMLProviders
15. GetSAMLProvider
16. CreateSAMLProviderMultipart
17. UpdateSAMLProviderRequest
18. ServeMetadata
19. ServeSigningCertificate
20. SAMLLoginHandler
21. SAMLCallbackHandler
22. SSOLoginHandler
23. SSOCallbackHandler

## Motivation and Context

Addresses [BED-5641](https://specterops.atlassian.net/browse/BED-5641)

This PR replaces integration tests with focused unit contract tests, making our test suite more maintainable and reliable. Contract tests provide better coverage of edge cases while being faster and less brittle than E2E tests.

## How Has This Been Tested?

All tests were run locally and verified to pass. Each test covers multiple scenarios including error cases and happy paths where applicable. The tests use mock objects to isolate the handlers from external dependencies.

## Types of changes

- Chore (a change that does not modify the application functionality)

## Checklist:

- [X] I have met the contributing prerequisites
  - Assigned myself to this PR
  - Added the appropriate labels
  - Associated an issue: https://github.com/SpecterOps/BloodHound/issues/672
  - Read the Contributing guide: https://github.com/SpecterOps/BloodHound/wiki/Contributing
- [X] I have ensured that related documentation is up-to-date
  - Open API docs
  - Code comments (GoDocs / JSDocs)
- [X] I have followed proper test practices
  - Added/updated tests to cover my changes
  - All new and existing tests passed

### CODE REVIEW

**Comment by kpowderly** [comment]
@coderabbitai full review

File: cmd/api/src/api/v2/collectors_test.go [hunk 1]
```diff
@@ -14,256 +14,143 @@
...
+					Method: http.MethodGet,
 				}
-
-				return request
 			},
-			createCollectorFile: func(t *testing.T) *os.File {
-				return nil
+			setupMocks: func(t *testing.T, mock *mock) {
+				mock.mockFS.EXPECT().ReadFile("InvalidCollectorType").Return([]byte{}, nil)
 			},
```
[the remark this answers is not in the document]
Comment (kpowderly) [reply cmd/api/src/api/v2/collectors_test.go]: @superlinkx I need to discuss the intended behavior of this when you get a chance today :)

File: cmd/api/src/api/v2/auth/sso_test.go [hunk 2]
```diff
@@ -296,3 +362,302 @@ func TestManagementResource_SanitizeAndGetRoles(t *testing.T) {
...
+			assert.Equal(t, testCase.expected.responseCode, status)
+			assert.Equal(t, testCase.expected.responseHeader, header)
+			if body != "" {
+				assert.JSONEq(t, testCase.expected.responseBody, body)
+			} else {
+				assert.Equal(t, testCase.expected.responseBody, body)
+			}
+		})
+	}
+}
```
[the remark this answers is not in the document]
Comment (kpowderly) [reply cmd/api/src/api/v2/auth/sso_test.go:663]: Resolving because success cases are handled separately already for SAML & OIDC

**Review by superlinkx — APPROVED** [review]
Absolutely excellent, I especially appreciate all the little details of validating the QR code parameters!
